File: react-frontend/src/services/AdminService.js
import axios from 'axios';

const API_URL = 'http://127.0.0.1:8080/admin';

class AdminService{
    getAdmins(){
        return axios.get(API_URL);
    }

    getAdminById(id){
        return axios.get(API_URL + '/' + id);
    }

    createAdmin(admin){
        return axios.post(API_URL, admin);
    }

    updateAdmin(admin, id){
        return axios.put(API_URL + '/' + id, admin);
    }

    deleteAdmin(id){
        return axios.delete(API_URL + '/' + id);
    }
}

export default new AdminService();

File: springboot-backend/src/main/java/com/example/student/loan/service/AdminService.java
package com.example.student.loan.service;

import com.example.student.loan.model.Admin;
import com.example.student.loan.model.AdminDTO;
import com.example.student.loan.repository.AdminRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.ArrayList;
import java.util.List;

@Service
public class AdminService {
    private final AdminRepository adminRepository;

    public AdminService(AdminRepository adminRepository) {
        this.adminRepository = adminRepository;
    }

    public List<Admin> getAllAdmins() {
        return adminRepository.findAll();
    }

    public AdminDTO getAdminDTOById(Integer id) {
        Admin admin = adminRepository.findById(id).orElse(null);
        if(admin == null) return null;
        return AdminDTO.createDTO(admin);
    }

    public Admin getAdminById(Integer id) {
        return adminRepository.findById(id).orElse(null);
    }

    public Admin createAdmin(Admin admin) {
        return adminRepository.save(admin);
    }

    public Admin updateAdmin(Integer id, Admin admin) {
        admin.setAdminID(id);
        return adminRepository.save(admin);
    }

    public void deleteAdmin(Integer id) {
        adminRepository.deleteById(id);
    }
}
